### ISSUE TITLE
change id to autoincrement in db, rather than config

### ISSUE THREAD

**Issue by etosch** [etosch/DB_Loader#6]
Current issue : if there are multiple concurrent loads, we run the risk of overlapping keys (though this doesn't seem to happen when there are concurrent loads from processes accessing the same config file - I had been concerned about multiple processes trying to access and update the same config file on one machine. db_loader had previously required a lock to write to .db_config. This worked fine locally, but when I tried running on fly, the process hung. When I tried just running concurrent scripts, it seemed that one process blocked access to the config file. This matters in cases where the write to the csv file is distributed over several nodes. Now my concern is for multiple systems (e.g. concurrent runs on swarm as well as fly) writing the same primary key.

My thought was to autoincrement the id, but now we run into the problem of matching the ids correctly with the values in the experiment and generations table. This would not be a problem if we were inserting either at the end of every run or continuously throughout a run. Instead, we are staging the runs in the csv files and only loading them when the size of the largest file approaches some value less than or equal to one gig. This lowers the cost of using the cloud and speeds things up.

For now I am going to try to grab the max value currently in the database and "reserve" space for inserts. That is, I will count the number of experiments to be loaded and just put a fake record in to bump up the maxid.

**Comment by etosch** [comment]
Current issue : if there are multiple concurrent loads, we run the risk of overlapping keys (though this doesn't seem to happen when there are concurrent loads from processes accessing the same config file - I had been concerned about multiple processes trying to access and update the same config file on one machine. db_loader had previously required a lock to write to .db_config. This worked fine locally, but when I tried running on fly, the process hung. When I tried just running concurrent scripts, it seemed that one process blocked access to the config file. This matters in cases where the write to the csv file is distributed over several nodes. Now my concern is for multiple systems (e.g. concurrent runs on swarm as well as fly) writing the same primary key.

My thought was to autoincrement the id, but now we run into the problem of matching the ids correctly with the values in the experiment and generations table. This would not be a problem if we were inserting either at the end of every run or continuously throughout a run. Instead, we are staging the runs in the csv files and only loading them when the size of the largest file approaches some value less than or equal to one gig. This lowers the cost of using the cloud and speeds things up.

For now I am going to try to grab the max value currently in the database and "reserve" space for inserts. That is, I will count the number of experiments to be loaded and just put a fake record in to bump up the maxid.
